src/CompareClasses.php

```diff
@@ -20,6 +21,7 @@ public function __construct(
         private ClassBased $classBasedComparisons,
         private InterfaceBased $interfaceBasedComparisons,
         private TraitBased $traitBasedComparisons,
+        private EnumBased|null $enumBasedComparisons = null, // optional argument for now to avoid BC break
```

Comment:
I'd suggest merging these into `ClassBased`: that removes all code changes from `Compareclasses`

Reply:
Done.